src/transaction.ts

```diff
@@ -52,13 +52,13 @@ export function assembleTransaction(
     );
   }
 
-  const coalesced = parseRawSimulation(simulation);
-  if (!coalesced.result) {
-    throw new Error(`simulation incorrect: ${JSON.stringify(coalesced)}`);
+  let success = parseRawSimulation(simulation);
```

Comment:
minor nit, your call, it's not success yet, maybe `simulationResponse`

Reply:
but it should be! heh. slightly more annoying to read later lines if it's something like `parsed`, and reassigning after the `if (!isSuccess...)` check is also slightly more annoying :upside_down_face: